[INFRA-2128] Rewrite the location base on the browse language

LinuxSuRen opened the pull request:
We have already translate all main pages of jenkins.io from English to Chinese.
If there is a better solution, please let know.

Commit:
Rewrite the location base on the browse language

olblak commented:
Awesome @LinuxSuRen, I'll test this PR :+1: before merging it

LinuxSuRen commented:
I’v tested this in my own nginx server. But not with current repo.

> On Jun 3, 2019, at 14:36, Olblak <notifications@github.com> wrote:
> 
> Awesome @LinuxSuRen <https://github.com/LinuxSuRen>, I'll test this PR 👍 before merging it
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/jenkins-infra/jenkins-infra/pull/1308?email_source=notifications&email_token=AALCFPKNSRT5PIJ2MYNYBATPYS3V3A5CNFSM4HR3IUC2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWYN7WI#issuecomment-498130905>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AALCFPORHB2RPT7GRIZO62TPYS3V3ANCNFSM4HR3IUCQ>.
>

olblak commented:
My first concern right now is to only redirect default traffic from jenkins.io/ to jenkins.io/zh and still allow readers to visit the english version so it would be better to limit the rewrite rule to '/' only

olblak committed:
Merge branch 'staging' into patch-1

LinuxSuRen commented:
Yes, I agree with you.

olblak committed:
Restrict language redirection to '/'

Update redirection url from https://jenkins.io/$lang/$1 to https://jenkins.io/$lang; in order to avoid url like jenkins.io// when we don't use lang=zh.

olblak committed:
Only rewrite jenkins.io '/' when using non default lang

Only rewrite jenkins.io/ when using non default lang in order to return a RC 200.

olblak reviewed (approved):
After few modiffication, this PR looks ready to be deployed. 
Just as a clarification, the purpose of this PR is to only redirect jenkins.io to jenkins.io/zh only if a browser is configured with local zh and only if visit the root website . in other work jenkins.io/bla is not redirected to jenkins.io/zh/bla

LinuxSuRen commented:
Thanks, @olblak .  I believe it's fine now.

daniel-beck commented:
This feature prevents users with `zh-CN` Accept-Language from ever using jenkins.io unless they land directly on a different page than the homepage. While this is limited to the homepage, there's no way to go back. Using the language switcher in the menu bar on the Chinese site simply doesn't work. And while other pages are unaffected, how would people even get there? From a search engine? That seems like a crappy UX.

Are there plans to address this? Otherwise I don't see why we should keep this automatic language redirect around.

LinuxSuRen commented:
Hi Daniel,

I've noticed this. I could try to fix this. But cannot promise a timeline.

Best,
Rick

On Sun, Oct 27, 2019 at 12:55 AM Daniel Beck <notifications@github.com>
wrote:

> This feature prevents users with zh-CN Accept-Language from ever using
> jenkins.io unless they land directly on a different page than the
> homepage. While this is limited to the homepage, there's no way to go back.
> Using the language switcher in the menu bar on the Chinese site simply
> doesn't work. And while other pages are unaffected, how would people even
> get there? From a search engine? That seems like a crappy UX.
>
> Are there plans to address this? Otherwise I don't see why we should keep
> this automatic language redirect around.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkins-infra/jenkins-infra/pull/1308?email_source=notifications&email_token=AALCFPOB4ILQP5ZPC4NIWHLQQRZA3A5CNFSM4HR3IUC2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECKMIHI#issuecomment-546620445>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALCFPMRI5BIZZA2222PIB3QQRZA3ANCNFSM4HR3IUCQ>
> .
>


-- 
Zhao Xiaojie (Rick)
Blog: https://github.com/LinuxSuRen
Twitter: https://twitter.com/suren69811254

daniel-beck commented:
@LinuxSuRen I saw https://github.com/jenkins-infra/cn.jenkins.io/issues/238 and removed my comment. Thanks.
